### ISSUE TITLE
OSD Overlay Rendering Issue (Mac Chrome)

### ISSUE THREAD

**Issue by vicewize** [fpv-wtf/wtfos-configurator#308]
Hey guys I was testing my old font and was just about to test some INav ones and more BF ones I finally got around to making and having this weird issue when rendering via online overlay tool. (Using KNIFA's dev branch) I fear its a Mac related issue, Im running a 2022 non M1/M2 16" Macbook Pro Maxed out on everything spec wise, I tried multiple other browsers and before I could use Opera but now it wont even let me render it just says "encoder init issue" or something at the top. I haven't tried it yet in Parallels or VMware yet but thats my next step. I have been watching  the devtools in the browswer and it is showing some odd things. Im sure prob Chrome.. But heres the original file and osd file and the post rendered single file. It will play about 2-3 sec then just lock up. Any help appreciated and no rush I know you guys are busy handling lots of other stuff. It will sort of playback in Premiere but then trys substituting frames and I even tried rendering it out to ProRes but no go. Again thanks!

Original DJI Vista DVR From Googles:

https://www.dropbox.com/s/bfkd9dwnkizon74/DJIG0336.mp4?dl=0

Font:

https://www.dropbox.com/scl/fo/64nud6s2j9619hueu0n0p/h?dl=0&rlkey=wmrkmhrbonld84cvnsu3ix7ut

Post Rendered File From Online Tool:

https://www.dropbox.com/s/1sn4gp9zg6d4uct/DJIG0335-osd.mp4?dl=0

_matt renshaw

**Comment by stylesuxx** [comment]
Could you also upload the OSD file?
It would also help if you could post the specific error you are getting, maybe a screenshot from the dev console too?

**Comment by vicewize** [comment]
Oops my bad here you go.. 

OSD File:

https://file.io/Yi2o5O7E0bQB

**Comment by stylesuxx** [comment]
Works fine on my end. Could you try on [develop](https://develop.fpv.wtf/osd-overlay) please? Technically I think, it should be on pretty much the same level as Knifa's branch - just to make sure.

Also please post the exact error you are running into.

**Comment by vicewize** [comment]
Its possible i selected the wrong one, again apologies, was up late. Let me check it out again. Seams to work fine in a windows environment on my end in chrome just not on my my macbook. Very odd. Heres the correct OSD file. Does the rendered video play fine for you as is or no? If so then could be some mac related GPU/driver issues.. Ughh Thanks btw!

Correct OSD File:

https://file.io/eGxweEcvsdzl

**Comment by stylesuxx** [comment]
Yes, plays fine for me. But there have been problems reported on Mac before. I was under the impression that this has been fixed in the meantime. Please also post the error messages you are getting, since this will likely help us the most in drilling down what the issue might be.

**Comment by vicewize** [comment]
Dont know if this helps at all but I was having the console open while it was running and theres 2 errors (they may be nothing as far as I know). :-)

<a href="https://ibb.co/HPXbWYT"><img src="https://i.ibb.co/8xXfW2g/html-console.jpg" alt="html-console" border="0"></a>

**Comment by stylesuxx** [comment]
Hmm you mentioned in the initial post you are getting an error regarding the encoder? The errors in the console are not really related to the encoding process. Also encoding seems to work fine, I mean, the bar finishes - right? Just the output is no good?

**Comment by vicewize** [comment]
Yes that was only happening with opera though, which was strange. If you need me to do anything specific i can.

**Comment by stylesuxx** [comment]
You have this with every video file? Because you uploaded different source and result files - now I am wondering if the issue is only with the 335 file?

**Comment by vicewize** [comment]
Heres a screengrab from opera I even went deep into the setttings to make sure hardware acceleration was enabled (dont known if that even makes a difference). 

<a href="https://ibb.co/D9kvvKT"><img src="https://i.ibb.co/HtYbbg3/1234.jpg" alt="1234" border="0"></a>

**Comment by vicewize** [comment]
Yeah i just tried it with another the following file 336 same thing, its driving me nuts. Freaking macs. It worked great until they pushed the last update out. They are always breaking things.

**Comment by vicewize** [comment]
No worries if its a deep dive issue, I know it can be probably 100 different things and you got to delve deep into it, I appreciate it regardless. For now I can use parallels just a bit slower is all.

**Comment by ripperhack** [comment]
@vicewize I found the "solution" ... with final cut just cut the last 3 seconds of the video then the render works. With DaVinci just cut where the message appears offline media which are precisely the last 3 or 4 final seconds

**Comment by stylesuxx** [comment]
Very interesting. And it have to be a couple seconds? Just the last frame for example is not enough?

**Comment by ripperhack** [comment]
Maybe a couple of frames is enough. I use DaVinci and in the timeline just after the flight statistics it appears "media offline", I cut exactly to the frame where there is the warning and solve the problem. With FinalCut not having the writing "media offline" and cut immediately after the statistics. If you watch the clip there is a part after that I don't understand what it is and it shows black. At the moment I'm at work, if I find a clip I'll upload a photo.

**Comment by ripperhack** [comment]
first image - media library, second image timeline
<img width="229" alt="first" src="https://user-images.githubusercontent.com/86116662/227133087-f454e6d2-c8f7-4acf-90e2-c9765b0ef4f3.png">
<img width="866" alt="second" src="https://user-images.githubusercontent.com/86116662/227133093-b6330c19-357b-4f83-89ab-8a100d89c673.png">

**Comment by ripperhack** [comment]
There seems to be some extra information in the osd file. Interestingly the file size with osd is smaller, in this example the original file without osd is 856MB while with osd it is 848MB. I don't know if it's normal I can't try it on windows.

**Comment by sandstreamer** [comment]
I've had a similar problem where the output OSD video imported to DaVinci Resolve causes "media offline" issue.
v2.1.0 - Trace ID: 58cbaf4a-dbc6-4dbb-a7bd-24fcc9b030c1
Chroma Key option was ticked.
M1 Macbook Pro 2021 (macOS 12.6.3)
Chrome 111.0.5563.110 (arm64)

A workaround that worked for me is to re-encode the "OSD" video file to constant framerate like this:
`ffmpeg -i input_video.mp4 -vsync cfr -c:a copy output_video.mp4`
Then the issue disappeared in DaVinci Resolve. 
(-c:a copy can be skipped I think since there is no audio)

**Comment by stylesuxx** [comment]
@sandstreamer so your issue was after rendering the file? All others seem to have issues while rendering the file. Do you still have the file before you converted it and could you share it please?

**Comment by stylesuxx** [comment]
@sandstreamer that would be great, just ping me on discord.

You are right - I then misunderstood the comment the first time around. I thought they cut off some frames from the end of the source file before rendering the OSD onto it.

**Comment by ripperhack** [comment]
I confirm, the problem occurs after rendering. When importing the file (with osd overlay) on davinci it has the media offline error. I solve it by cutting a few frames at the end of the file (with osd overlay) just imported
